server/src/main/java/org/elasticsearch/cluster/SafeClusterStateSupplier.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.cluster;
+
+import java.util.Optional;
+
+import static org.elasticsearch.gateway.GatewayService.STATE_NOT_RECOVERED_BLOCK;
+
+/**
+ * Utility to access {@link ClusterState} only when it is "ready", where "ready" means initialized and recovered.
```

Comment:
I like that you defined `ready` here!

Could you `@link` to `isInitialized` and `STATE_NOT_RECOVERED_BLOCK` for folks who want to know more about what "initialized" and "recovered" mean?

Reply:
I added details to the javadoc